### PR TITLE
Re-implement Printer as a class based module

### PR DESCRIPTION
**Pull request by Jellyfrog**
Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: LibreNMS/Interfaces/Polling/PrinterPolling.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+namespace LibreNMS\Interfaces\Polling;
+
+interface PrinterPolling
+{
+    /**
+     * @return
+     */
+    public function pollPrinter();
```
Comment (murrant) [inline LibreNMS/Interfaces/Polling/PrinterPolling.php:30]: should send it the models that need to be polled.

Comment (Jellyfrog) [reply LibreNMS/Interfaces/Polling/PrinterPolling.php:30]: These files are actually left-overs from the none implementation of os-overrides, should I implement it or delete them? :)

Comment (murrant) [reply LibreNMS/Interfaces/Polling/PrinterPolling.php:30]: ah, go ahead and delete.  Easy to re-add if needed.

**Comment by murrant** [comment]
I was thinking about the state sensors.
It kind of solves the issue if this module is called "PrinterSupplies"  Then the state sensors should not be here.

File: database/migrations/2021_03_11_003713_rename_printer_columns.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+class RenamePrinterColumns extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('printer_supplies', function (Blueprint $table) {
```
Comment (murrant) [inline database/migrations/2021_03_11_003713_rename_printer_columns.php:16]: You can do these all in the same Schema::table block

File: includes/html/pages/device/overview/toner.inc.php [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 $graph_type = 'toner_usage';
 
-$supplies = \App\Models\Toner::query()->where('device_id', $device['device_id'])->get()->groupBy('toner_type');
+$supplies = \App\Models\PrinterSupply::query()->where('device_id', $device['device_id'])->get()->groupBy('printer_type');
```
Comment (murrant) [inline includes/html/pages/device/overview/toner.inc.php:7]: ->groupBy('printer_type');  this doesn't make much sense.  why did you prefix the fields with printer_?

**Review by murrant — COMMENTED** [review]
We will need a hook in config to populate printer-supplies module discovery and poller options  from the existing toner options.

File: database/migrations/2021_03_11_003713_rename_printer_columns.php (hunk 2 above)
Comment (Jellyfrog) [reply database/migrations/2021_03_11_003713_rename_printer_columns.php:16]: Nope, sqlite doesnt support it.

File: includes/html/pages/device/overview/toner.inc.php (hunk 3 above)
Comment (Jellyfrog) [reply includes/html/pages/device/overview/toner.inc.php:7]: it does makes sense since the data is grouped by it :)

**Review by murrant — APPROVED** [review]
Looks great.

**Pull request merged**